File: Lab5/StringLibDLL/BigLetters.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StringLibDLL
{
    public class BigLetters : Strings
    {
        private string data;

        public BigLetters(string str)
        {
            data = str;
        }

        public override int Length()
        {
            return (Encoding.UTF8.GetByteCount("B") * SymbCount());
        }

        public override int SymbCount()
        {
            int count = 0;
            foreach (char c in data)
            {
                if (c == 'B') count++;
            }
            return count;
        }
    }
}

File: Lab5/StringLibStatic/BigLetters.cpp
#include "BigLetters.h"

BigLetters::BigLetters(string str)
{
    data = str;
}

int BigLetters::Length()
{
    return (sizeof('B') * SymbCount());
}

int BigLetters::SymbCount()
{
    int count = 0;
    for (char c : data)
    {
        if (c == 'B') count++;
    }
    return count;
}